modules/core/src/norm.cpp

```diff
@@ -1243,9 +1243,9 @@ double norm( InputArray _src1, InputArray _src2, int normType, InputArray _mask
 
     if( normType == NORM_INF )
     {
-        if (depth == CV_64F || depth == CV_16F)
+        if (depth == CV_64F)
             return result.d;
-        else if (depth == CV_32F)
+        else if (depth == CV_32F || depth == CV_16F)
```

Comment:
Why? See
https://github.com/opencv/opencv/blob/cf0ba039c3cfcd2aabdcd060a1ca4fc79515975d/modules/core/src/norm.cpp#L1227

Reply:
Seems like it works because `resuls` is `union`:
```cpp
union
    {
        double d;
        float f;
        int i;
        unsigned u;
    }
```